modules/02_welfare/utilitarian/equations.gms

```diff
@@ -21,21 +21,57 @@ q02_welfareGlob..
-        pm_welf(ttot) * pm_ts(ttot) * (1 / ( (1 + pm_prtp(regi))**(pm_ttot_val(ttot)-2005) ) )
-        *   (  (pm_pop(ttot,regi) 
-                *   (
-                        ((( (vm_cons(ttot,regi))/pm_pop(ttot,regi))**(1-1/pm_ies(regi))-1)/(1-1/pm_ies(regi)) )$(pm_ies(regi) ne 1)
-                       + (log((vm_cons(ttot,regi)) / pm_pop(ttot,regi)))$(pm_ies(regi) eq 1)
-                    )
+      pm_welf(ttot)
+    * pm_ts(ttot)
+    * (1 / ((1 + pm_prtp(regi)) ** (pm_ttot_val(ttot) - 2005)))
+    * ( ( pm_pop(ttot,regi) 
+        * (
+            ( ( ( vm_cons(ttot,regi)
+                / pm_pop(ttot,regi)
                 )
-$if %cm_INCONV_PENALTY% == "on"  - v02_inconvPen(ttot,regi) - v02_inconvPenCoalSolids(ttot,regi)
-$if "%cm_INCONV_PENALTY_FESwitch%" == "on"  - sum((entySe,entyFe,te,sector,emiMkt)$(se2fe(entySe,entyFe,te) AND entyFe2Sector(entyFe,sector) AND sector2emiMkt(sector,emiMkt) AND (entySeBio(entySe) OR entySeSyn(entySe) OR entySeFos(entySe)) ), v02_NegInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt) + v02_PosInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt))/1e3	
-            )
+             ** (1 - (1 / pm_ies(regi)))
+              - 1
+              )
+            / (1 - (1 / pm_ies(regi)))
+            )$( pm_ies(regi) ne 1 )
+          + log((vm_cons(ttot,regi)) / pm_pop(ttot,regi))$( pm_ies(regi) eq 1 )
+          )
         )
+$ifthen %cm_INCONV_PENALTY% == "on"
+      - v02_inconvPen(ttot,regi)
+      - v02_inconvPenCoalSolids(ttot,regi)
+$endif
+$ifthen "%cm_INCONV_PENALTY_FESwitch%" == "on"
+        !! inconvenience cost for fuel switching in FE between fossil,
+        !! biogenic, synthetic solids, liquids and gases across sectors and
+        !! emissions markets
+      - sum((entySe,entyFe,te,sector,emiMkt)$(    
+                                                se2fe(entySe,entyFe,te)
+                                            AND entyFe2Sector(entyFe,sector)
+                                            AND sector2emiMkt(sector,emiMkt)
+                                            AND (   entySeBio(entySe) 
+                                                 OR entySeSyn(entySe)
+                                                 OR entySeFos(entySe))       ),
+          v02_NegInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt)
+        + v02_PosInconvPenFeBioSwitch(ttot,regi,entySe,entyFe,sector,emiMkt)
+        )
+      / 1e3	
+        !! inconvenience cost for fuel switching in FE non-energy use between biogenic, synthetic solids, liquids and gases
+      - sum((entySe,entyFe,te,sector,emiMkt)$(
+                            se2fe(entySe,entyFe,te) 
+                        AND entyFe2sector2emiMkt_NonEn(entyFe,sector,emiMkt)
+                        AND (entySeBio(entySe) OR entySeSyn(entySe) )        ),
+          v02_NegInconvPenNonEnSwitch(ttot,regi,entySe,entyFe,sector,emiMkt) 
+        + v02_NegInconvPenNonEnSwitch(ttot,regi,entySe,entyFe,sector,emiMkt)
```

Comment:
Should not one of these be positive?

Reply:
I'm running tests now to investigate effect of fixing this